mipengine/node/tasks/udfs.py

```diff
@@ -499,6 +498,7 @@ def _generate_udf_statements(
             "udf_name": udf_name,
             "node_id": node_config.identifier,
             "min_row_count": node_config.privacy.minimum_row_count,
+            "request_id": request_id,
```

Comment:
This is mostly a refactoring PR. i.e. it doesn't change functionality or API. What you say requires changes in the generator API. 

E.g. the generator could be an object implementing methods `get_definition` and `get_select_statement` where `udf_name`, `node_id` etc are passed as parameters. That would remove the need to use strings to replace stuff in templates. 

But this is out of scope for the current PR and requires much more work.

Reply:
Why don't we just have these strings `udf_name`, `node_id` ...  in the udfgen and import them from there? Instead of hardcoding them in 2 different places, once in the udfgenerator and once in the udfs? Is it that much work?

Currently, `request_id` is hardcoded in the `ast.py` file, `udf_name` is hardcoded in the `udfgenerator.py` and `min_row_count` is hardcoded in the `iotypes.py` ... We just have to import them from the same place.